### PR TITLE
Add name functionality for comments

### PR DESCRIPTION
**Pull request by shandyk18**
Names can be added to comments to identify the author of a comment.

A new text area was added in the comment section. In doGet(), the name and comment are stored in a LInkedHashMap to maintain insertion order.

Screenshot 2020-07-01 at 3 15 11 PM - Display 2

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java [hunk 1]
```diff
@@ -45,14 +47,16 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     int maxComments = maxString == null ? 5 : Integer.parseInt(maxString);
     int counter = 0;
 
-    List<String> commentHistory = new ArrayList<>();
+    Map<String, String> commentHistory = new LinkedHashMap<>();
```
Comment (wujingya) [inline portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:50]: LinkedHashMap doesn't allow duplicate keys, are you intentionally disallow comments with the same name?

Comment (shandyk18) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:50]: Do you have a recommendation on how to allow duplicate keys? Would using a MultiMap be a good idea?

Comment (shandyk18) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:50]: I could also have an ArrayList of values for each key to allow for multiple comments?

**Comment by shandyk18** [comment]
Comments from the same user is now possible. 
The map now contains a String key and an ArrayList of comments. 

Screenshot 2020-07-06 at 11 13 33 AM

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java (hunk 1 above)
Comment (wujingya) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java]: Maybe we can have a Comment class that holds name and comment, and then still use an ArrayList here?

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java [hunk 2]
```diff
@@ -45,14 +47,20 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     int maxComments = maxString == null ? 5 : Integer.parseInt(maxString);
     int counter = 0;
 
-    List<String> commentHistory = new ArrayList<>();
+    //Multimap<String, String> multimap = ArrayListMultimap.create();
```
Comment (MorganAw) [inline portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:50]: prefer to delete commented out code before submission

Comment (MorganAw) [inline portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:50]: prefer to delete commented out code before submission

Comment (shandyk18) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:50]: Regarding the feature, would that replace the number of comments selection?

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java (hunk 1 above)
Comment (shandyk18) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java]: Would you still recommend making a separate class for a Comment with the way I have it implemented now? I created a separate class (that hasn't been pushed) but there doesn't seem to be much difference in code for the servlet.

Screenshot 2020-07-06 at 12 26 04 PM - Display 2

Comment (wujingya) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java]: Sure what you currently have works too, sorry I missed it last time.

**Pull request merged**